js/stateless.js/src/state/utxo.ts

```diff
@@ -0,0 +1,228 @@
+import { PublicKey } from "@solana/web3.js";
+import { TlvDataElement, createTlvDataElement } from "./utxo-data";
+import { bigint254, bn, createBigint254 } from "./bigint254";
+import { BN } from "@coral-xyz/anchor";
+
+/// TODO: implement PublicKey254 type based on bigint254
+/// figure which fmt to use in indexer and on client side: since solana's regluar 'PublicKey' expects padding to 32.
+export type PublicKey254 = PublicKey;
+
+/** Describe the generic utxo details applicable to every utxo. */
+export type Utxo = {
+  /** Public key of program or user that owns the utxo */
+  owner: PublicKey;
+  /** Optional lamports attached to the utxo */
+  lamports: number | BN;
+  /** Optional data attached to the utxo */
+  data: TlvDataElement[] | null;
+  /**
+   * TODO: Implement address functionality
+   * Optional unique account ID that is persistent across transactions.
+   */
+  address?: PublicKey;
+};
+
+/** Context for utxos inserted into a state Merkle tree */
+export type MerkleContext = {
+  /** Poseidon hash of the utxo preimage  */
+  hash: bigint254;
```

Comment:
could consider it renaming to leaf since we have leafIndex below.

Reply:
adding into inline comment for now, will ask helius for thoughts since this will affect rpc interface